src/pages/smart-contracts/clarity-values-guide.md

```diff
@@ -0,0 +1,143 @@
+---
+title: Clarity Values Guide
+description: Learn how to deal with Clarity Values in JavaScript.
+tags:
+  - tutorial
+---
+
+## Introduction
+The Clarity language makes use of a strong static [type system](https://docs.blockstack.org/references/language-clarity#clarity-type-system). This simply means that every function defined in Clarity expects arguments of specific types, and that a failure to provide arguments of the specified types will result in your code failing to compile, or your `contract-call` transaction failing prior to execution.
+
+In order to build web applications that interact with Clarity contracts, we must include type information as a part of our function calls. The [@stacks/transactions](https://github.com/blockstack/stacks.js/tree/master/packages/transactions) library makes this easy, as we will demonstrate below.
+
+## Clarity Types
+The Clarity type system includes the following types:
+
+- `(tuple (key-name-0 key-type-0) (key-name-1 key-type-1) ...)` - a typed tuple with named fields.
+- `(list max-len entry-type)` - a list of maximum length max-len, with entries of type entry-type
+- `(response ok-type err-type)` - object used by public functions to commit their changes or abort. May be returned or used by other functions as well, however, only public functions have the commit/abort behavior.
+- `(optional some-type)` - an option type for objects that can either be (some value) or none
+- `(buff max-len)` - byte buffer or maximum length max-len.
+- `(string-ascii max-len)` - ASCII string of maximum length max-len
+- `(string-utf8 max-len)` - UTF-8 string of maximum length max-len
+- `principal` - object representing a principal (whether a contract principal or standard principal).
+- `bool` - boolean value (true or false)
+- `int` - signed 128-bit integer
+- `uint` - unsigned 128-bit integer
+
+## Constructing Clarity Values
+Clarity values can be constructed with functions provided by the **@stacks/transactions** library. These function simply output javascript objects that contain a value and a numerical representation of the Clarity type information (or in the case of Booleans and Optional None, just the type information). These Clarity value objects can then be easily serialized into `contract-call` transactions for interacting with Clarity contracts.
```

Comment:
`These functions [...]`

Reply:
"contract-call transactions" > "contract call transactions" (assuming the reader does not yet know that is the type (might be a bit confusing). I'd also reference something like this chapter: https://docs.blockstack.org/stacks-blockchain/transactions#construction

Reply:
The functions are listed directly below